.github/workflows/build-publish.yml

```diff
@@ -7,10 +7,6 @@
 name: build-publish
 
 on:
-  push:
-    branches: [main]
-  pull_request:
-    branches: [main]
```

Comment:
Should we keep this for push on main? @elliottower 
I noticed it because we don't have anymore build-publish on merged PRs

Reply:
Oh I guess gymnasium also does it for all prs I guess it’s just “can this be built” and only uploads if it’s a release. My bad about that, I don’t think it’ll be a huge deal but probably best to be consistent. Pettingzoo it was just for release but since this is much more based on gymnasium obviously we should inherit from there